src/lib/types/src/Flow/Types/Type/Native/String/StringTypeNarrower.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Flow\Types\Type\Native\String;
+
+use function Flow\Types\DSL\{type_html, type_json, type_uuid, type_xml};
+use Flow\Types\Type;
+use Flow\Types\Type\Native\StringType;
+use Flow\Types\Type\TypeNarrower;
+
+final readonly class StringTypeNarrower implements TypeNarrower
+{
+    public function narrow(Type $type, mixed $value) : Type
+    {
+        if (!$type instanceof StringType || !\is_string($value)) {
+            return $type;
+        }
+
+        $checker = new StringTypeChecker($value);
```

Comment:
The idea was to rename StringTypeChecker into `StringTypeNarrower` and apply the TypeNarrower on it